src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Initialization.cs

```diff
@@ -27,11 +31,36 @@ static OpenSsl()
         }
     }
 
-    internal static partial class CryptoInitializer
+    internal static unsafe partial class CryptoInitializer
     {
-        static CryptoInitializer()
+        internal struct MemoryEntry
         {
-            if (EnsureOpenSslInitialized() != 0)
+            public char* File;
+            public int Size;
+            public int Line;
+        }
+
+        private static readonly bool DebugMemory = GetMemoryDebug("DOTNET_SYSTEM_NET_SECURITY_OPENSSL_MEMORY_DEBUG");
+        private static readonly bool ValidateMemory = GetMemoryDebug("DOTNET_SYSTEM_NET_SECURITY_OPENSSL_MEMORY_VALIDATE");
+        private static readonly IntPtr Offset = sizeof(MemoryEntry);
```

Comment:
This is going to be 12 on 32-bit systems. It may mess up required alignment. Does OpenSSL have any alignment requirements on the allocator?